### DISCUSSION TITLE
Armbian image have a different gcc version that the one used for kernel compilation

### DISCUSSION THREAD

**Discussion by xFall00** [ophub/amlogic-s9xxx-armbian#1778] [Amlogic]
Guys, I installed "Armbian_23.08.0_amlogic_s905x2_lunar_5.15.133_server_2023.10.02.img.gz" on SD card and entered into TESLA MediaTech X700 with soc s905x2 in it.

It boots flawlessly, I have created user name etc ...

I am trying to install xone driver. https://github.com/medusalix/xone

When I hit the "sudo ./install.sh --release" command, I got warning that my gcc is different the one that was used for kernel compilation. I belive that is the issue of not compiling the code.

Why is it different ? What can I do in order to match it ?

Here are details of the commnad:

`Installing xone v0.3-2-gbbf0dcc...
Sign command: /lib/modules/5.15.133-ophub/build/scripts/sign-file
Binary /lib/modules/5.15.133-ophub/build/scripts/sign-file not found, modules won't be signed
Creating symlink /var/lib/dkms/xone/v0.3-2-gbbf0dcc/source -> /usr/src/xone-v0.3-2-gbbf0dcc

Building module:
Cleaning build area...
make -j4 KERNELRELEASE=5.15.133-ophub -C /lib/modules/5.15.133-ophub/build M=/var/lib/dkms/xone/v0.3-2-gbbf0dcc/build.....(bad exit status: 2)
Error! Bad return status for module build on kernel: 5.15.133-ophub (aarch64)
Consult /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/make.log for more information.
DKMS make.log for xone-v0.3-2-gbbf0dcc for kernel 5.15.133-ophub (aarch64)
St 11. října 2023, 21:39:39 CEST
make: Entering directory '/usr/src/linux-headers-5.15.133-ophub'
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: aarch64-none-linux-gnu-gcc (Arm GNU Toolchain 12.3.Rel1 (Build arm-12.35)) 12.3.1 20230626
  You are using:           gcc (Ubuntu 12.3.0-1ubuntu1~23.04) 12.3.0
  CC [M]  /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/transport/wired.o
  CC [M]  /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/transport/dongle.o
  CC [M]  /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/transport/mt76.o
  CC [M]  /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/bus/bus.o
/bin/sh: 1: scripts/basic/fixdep: Exec format error
make[1]: *** [scripts/Makefile.build:289: /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/bus/bus.o] Error 126
make[1]: *** Deleting file '/var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/bus/bus.o'
make[1]: *** Waiting for unfinished jobs....
/bin/sh: 1: scripts/basic/fixdep: Exec format error
make[1]: *** [scripts/Makefile.build:289: /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/transport/wired.o] Error 126
make[1]: *** Deleting file '/var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/transport/wired.o'
/bin/sh: 1: scripts/basic/fixdep: Exec format error
make[1]: *** [scripts/Makefile.build:289: /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/transport/dongle.o] Error 126
make[1]: *** Deleting file '/var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/transport/dongle.o'
/bin/sh: 1: scripts/basic/fixdep: Exec format error
make[1]: *** [scripts/Makefile.build:289: /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/transport/mt76.o] Error 126
make[1]: *** Deleting file '/var/lib/dkms/xone/v0.3-2-gbbf0dcc/build/transport/mt76.o'
make: *** [Makefile:1909: /var/lib/dkms/xone/v0.3-2-gbbf0dcc/build] Error 2
make: Leaving directory '/usr/src/linux-headers-5.15.133-ophub'`

**Comment by ophub** [comment]
Update the kernel to the dev series and then recompile the drivers. There is information about this in the search history.

**Comment by xFall00** [comment]
ok solved. https://forum.armbian.com/topic/30699-recomended-armbian-version-for-soc-s905x2/. Basically I have used the related kernel 6.1.11, then installed the linux headers specific version related to the kernel, compiled the xone driver according to https://github.com/medusalix/xone, paired xbox series x or one x controller (had to be done on Windows), rebooted with USB dongle in, connected the controler(s) and it works.

Important part is to hold the respective linux-headers for the next apt update/upgrade. I have tried with kernel 6.1.11 (works), 6.1.30 and 6.1.50 (does not work) for my box.
